File: python/day8.py
input = open("..\input\day8.txt", "r").readline().rstrip()
min,res,width,height = 99999,0,25,6
output = ['2'] * (width*height)
layers = [input[i:i+(width*height)] for i in range(0, len(input), width*height)]
for layer in layers:
    if layer.count('0') < min:
        min=layer.count('0')
        res=layer.count('1') * layer.count('2')
    for x in range(width*height):
        if output[x]=='2':
            output[x]=layer[x]
print ("AnsewerA",res)
for line in range(height):
    p=''
    for c in output[(line*width):((line+1)*width)]:
        p = p + ' ' if c == '0' else p + '#'
    print(p)

File: aoc2019/day8.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace aoc2019
{
    public class day8
    {
        public static void Go()
        {
            string input = File.ReadLines(@"c:\github\aoc2019\input\day8.txt").First();
            int width = 25;
            int height = 6;
            string[] layers = Enumerable.Range(0, input.Length / (width * height)).Select(i => input.Substring(i * width * height, width * height)).ToArray();
            string thelayer = layers.Where(s => s.Count(f => f == '0') == layers.Min(s => s.Count(f => f == '0'))).First();
            Console.WriteLine(string.Format("AnswerA {0}", thelayer.Count(f => f == '1') * thelayer.Count(f => f == '2')));
            Console.WriteLine("AnswerB:");
            for (int i = 0; i < height; i++)
            {
                string line = "";
                for (int j = 0; j < width; j++)
                {
                    int pos = (width * i) + j;
                    foreach (string layer in layers)
                    {
                        if (layer[pos] != '2')
                        {
                            if (layer[pos] == '0')
                                line += ' ';
                            else
                                line += 'X';
                            break;
                        }
                    }
                }
                Console.WriteLine(line);
            }
        }
    }
}
